pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -532,6 +532,10 @@ private PGStream enableSSL(PGStream pgStream, SslMode sslMode, Properties info,
 
     LOGGER.log(Level.FINEST, " FE=> SSLRequest");
 
+    int sslTimeout = PGProperty.SSL_RESPONSE_TIMEOUT.getInt(info);
+    int currentTimeout = pgStream.getSocket().getSoTimeout();
```

Comment:
If the user already has a custom timeout that's less than this default (e.g. 2 seconds), this would have the side effect of raising the timeout during the SSL handshake.

How about something like this to take the min of the two?

```
int currentTimeout = pgStream.getSocket().getSoTimeout();
int sslTimeout = PGProperty.SSL_RESPONSE_TIMEOUT.getInt(info);
if (currentTimeout > 0 && currentTimeout < sslTimeout) {
    sslTimeout = currentTimeout;
}
```